### PR TITLE
*: Support user management through crd. #175

### LINKED ISSUE #150: [feature] create crd when install operator.
**Is your feature request related to a problem? Please describe.**

Take backup as an example：

`clusters.mysql.radondb.com` crd can automatically installed, but need install `backups.mysql.radondb.com` CRD manually.
operator helm: https://github.com/radondb/radondb-mysql-kubernetes/tree/main/charts

**Describe the solution you'd like**

install `clusters.mysql.radondb.com` and `backups.mysql.radondb.com` when install operator 

- add `mysql.radondb.com_backups.yaml` to `charts/mysql-operator/crds/`
- update `mysql.radondb.com_clusters.yaml`
- modify `charts/mysql-operator/templates/cluster_rbac.yaml`

**Describe alternatives you've considered**

**Additional context**

### LINKED ISSUE #175: [feature] Support user management through crd
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

Add, delete and modify users through user crd.

**Describe alternatives you've considered**

Extend kubectl. reference: https://github.com/kubernetes/sample-cli-plugin

**Additional context**

### PR DESCRIPTION
**Pull request by runkecheng**
### What type of PR is this?

/enhancement

### Which issue(s) this PR fixes?

Fixes #150 #175 #189 #220

Previous pr：#198 #218

### What this PR does?

Summary:

- [x] Support adding and deleting users
- [x] Support for user authorization (databases, tables)
- [x] Support modifying hosts
- [x] Support automatic installation of msyqluser crd
- [x] Add the verification of user parameters.

### Special notes for your reviewer?

The main logic is in this commit: 

For test:
1. Delete the original cluster and crd, and redeploy using the test image.
    - operator image: `radondb/mysql-operator:demo-921`
    - sidecar image: `radondb/mysql-sidecar:demo-921`
2. Make sure that the root password of the msyql cluster is set.
3. Make sure to create user secret before creating user crd.

### CODE REVIEW

**Comment by andyli029** [comment]
Another pr about `SQLRunnerFactory` .

File: controllers/mysqluser_controller.go [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+
+		// Update resource so it will remove the finalizer.
+		if err := r.Update(ctx, mysqlUser.Unwrap()); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *MysqlUserReconciler) reconcileUserInCluster(ctx context.Context, mysqlUser *mysqluser.MysqlUser) (err error) {
```
Comment (andyli029) [inline controllers/mysqluser_controller.go:125]: Add comment log.

File: internal/query.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+// Args representation of the query.
+func (q *Query) Args() []interface{} {
+	return q.args
+}
+
+// NewQuery returns a new Query object.
+func NewQuery(q string, args ...interface{}) Query {
+	if q == "" {
+		panic("unexpected empty query")
```
Comment (andyli029) [inline internal/query.go:41]: Fix about #218

**Comment by acekingke** [comment]
LGTM

File: utils/constants.go [hunk 3]
```diff
@@ -130,6 +130,8 @@ const (
 	ServiceAccount ResourceName = "service-account"
 	// PodDisruptionBudget is the name of pod disruption budget for the statefulset.
 	PodDisruptionBudget ResourceName = "pdb"
+	// UserControllerName is the name of UserController.
+	UserControllerName ResourceName = "controller.user"
```
Comment (andyli029) [inline utils/constants.go:134]: Delete it.

File: utils/common.go [hunk 4]
```diff
@@ -98,3 +98,25 @@ func BuildBackupName() string {
 	return fmt.Sprintf("backup_%v%v%v%v%v%v", cur_time.Year(), int(cur_time.Month()),
 		cur_time.Day(), cur_time.Hour(), cur_time.Minute(), cur_time.Second())
 }
+
+func StringDiffIn(actual, desired []string) []string {
+	diff := []string{}
+	for _, aStr := range actual {
+		// if is not in the desired list remove it
```
Comment (andyli029) [inline utils/common.go:105]: Add `.` to the end.

File: api/v1alpha1/mysqluser_types.go [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+	User string `json:"user,omitempty"`
+
+	// Hosts is the grants hosts.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems=1
+	Hosts []string `json:"hosts,omitempty"`
+
+	// ClusterBinder Contains parameters about the cluster bound by user.
+	// +kubebuilder:validation:Required
+	ClusterBinder ClusterBinder `json:"clusterBinder,omitempty"`
```
Comment (andyli029) [inline api/v1alpha1/mysqluser_types.go:45]: Better name about `ClusterBinder`

**Comment by andyli029** [comment]
LGTM.

**Pull request merged**